localstack-core/localstack/services/ec2/resource_providers/aws_ec2_subnet.py

```diff
@@ -41,6 +41,47 @@ class Tag(TypedDict):
 REPEATED_INVOCATION = "repeated_invocation"
 
 
+def generate_subnet_read_payload(
+    ec2_client, schema, subnet_ids: Optional[list[str]] = None
+) -> list[EC2SubnetProperties]:
```

Comment:
Suggestion: Seems like `generate_subnet_read_payload`  could be use it for both `list` and `read` like you did in other resource providers.

Reply:
ah yes! That was the goal. Forgot to update that one! Thanks for the catch! :pray: